R/parse_system.R

```diff
@@ -600,6 +656,14 @@ parse_system_arrays <- function(exprs, call) {
     vcapply(exprs[is_dim], function(eq) eq$lhs$name),
     dim_nms)
 
+  for (i in which(is_dim)) {
+    x <- exprs[[i]]
+    extra_dims <- unlist(lapply(x$lhs$args, deparse))
+    for (d in extra_dims) {
+      name_dim_equation[[d]] <- odin_dim_name(d)
+    }
```

Comment:
This is doing much more than needed:

* we already have the deparsed names above, and that variable (though I've suggested we can refactor it)
* odin_dim_name is vectorised already

this can certainly be done entirely as a single assignment replacing the call on 655. But have a look logistically and see if you can generally make this whole section less complicated. Consider again another function that just builds and returns name_dim_equation or something?